src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -264,7 +267,7 @@ public abstract partial class JsonNode
         public static System.Text.Json.JsonNode Parse(string json) { throw null; }
         public static bool TryGetNode(System.Text.Json.JsonElement jsonElement, out System.Text.Json.JsonNode jsonNode) { throw null; }
     }
-    public partial class JsonNumber : System.Text.Json.JsonNode
+    public partial class JsonNumber : System.Text.Json.JsonNode, System.IEquatable<System.Text.Json.JsonNumber>
```

Comment:
@bartonjs would it make sense to also make this implement IEquatable<int>, IEquatable<double>, etc.?

Reply:
> make this implement IEquatable, IEquatable, etc.?

Was one of them meant to be generic?

Reply:
Assuming that your IEquatables were generic over numeric value types:

Not really.

The equality for this is either very complicated, simple but maybe not useful.  JSON numbers are any legal value in `Q` (the set of rational (real) numbers).  If we're looking at this from a mathy approach then "3" "3e0", "30e-1", "3.0", "3.00", and the infinite (up to memory limits) other possible ways of being "3" are all equal.  I doubt the implementation is going to take that into account.  I expect that it's just "do the two implementations have the same string, internally?".

An `IEquatable<int>` implementation could be written that does TryParse on the field and matches against the input.  But is it useful?

As far as explicit guidelines, all we have are that immutable value types should be IEquatable of themselves, and you should consider being IEquatable of yourself if you override Object.Equals, and if you're IComparable of yourself you should be IEquatable of yourself.

We don't have one that says "don't be IEquatable of not-you", but instinctively (as a mathematician) that "do not" makes sense to me because the inverse relationship isn't true (int isn't equatable to JsonNumber, so JsonNumber shouldn't be equatable to int, by symmetry)

Reply:
> Was one of them meant to be generic

Yes sorry, codeflow didn't like the fact that I didn't use code blocks to define my generics 😆